x-pack/plugins/enterprise_search/public/applications/applications/components/engine/engine_connect/search_application_api.tsx

```diff
@@ -133,41 +188,104 @@ export const SearchApplicationAPI = () => {
+        <GenerateEngineApiKeyModal
+          engineName={searchApplicationName}
+          onClose={closeGenerateModal}
+        />
       ) : null}
       <EuiCallOut
         iconType="iInCircle"
         title={
           <FormattedMessage
-            id="xpack.enterpriseSearch.content.searchApplication.api.safeSearchCallout.title"
+            id="xpack.enterpriseSearch.content.searchApplication.safeSearchApi.safeSearchCallout.title"
             defaultMessage="What is Safe Search API?"
           />
         }
       >
         <FormattedMessage
-          id="xpack.enterpriseSearch.content.searchApplication.api.safeSearchCallout.body"
-          defaultMessage="The safe search API endpoint only allows queries with the parameters defined in the settings JSON, so you can create public-facing search endpoints for your Elasticsearch indices."
+          id="xpack.enterpriseSearch.content.searchApplication.safeSearchApi.safeSearchCallout.body"
+          defaultMessage="The {safeSearchDocumentation} allows you to create a secure, public-facing search endpoint for your search application. This endpoint only accepts queries that match the parameters defined in your {searchTemplateDocumenation}."
+          values={{
+            safeSearchDocumentation: (
+              <EuiLink
+                data-test-subj="safe-search-documentation-link"
+                href="#" // replace with safe search api doc link
+                data-telemetry-id="entSearchApplications-safeSearchApi-documentation-viewDocumentaion"
+              >
+                {i18n.translate(
+                  'xpack.enterpriseSearch.content.searchApplication.safeSearchApi.safeSearchCallout.body.safeSearchDocLink',
+                  {
+                    defaultMessage: 'Safe Search API',
+                  }
+                )}
+              </EuiLink>
+            ),
+            searchTemplateDocumenation: (
+              <EuiLink
+                data-test-subj="search-template-documentation-link"
+                href={docLinks.searchTemplates}
+                data-telemetry-id="entSearchApplications-searchTemplate-documentation-viewDocumentaion"
+              >
+                {i18n.translate(
+                  'xpack.enterpriseSearch.content.searchApplication.safeSearchApi.safeSearchCallout.body.searchTemplateDocLink',
+                  {
+                    defaultMessage: 'search template',
+                  }
+                )}
+              </EuiLink>
+            ),
+          }}
+        />
+        <EuiSpacer size="l" />
+        <FormattedMessage
+          id="xpack.enterpriseSearch.content.searchApplication.safeSearchApi.safeSearchCallout.safeSearchLearnMore"
+          defaultMessage="{safeSearchDocumentation}"
+          values={{
+            safeSearchDocumentation: (
+              <EuiLink
+                data-test-subj="safe-search-documentation-link"
+                href="#" // replace with safe search api doc link
```

Comment:
do we have this yet?

Reply:
We don't yet.  https://github.com/elastic/kibana/pull/159356#issuecomment-1591946615